jdaviz/utils.py

```diff
@@ -385,3 +391,127 @@ def total_masked_first_data(self):
     def __setgluestate__(cls, rec, context):
         masks = {key: context.object(value) for key, value in rec['masks'].items()}
         return cls(masks=masks)
+
+
+def download_uri_to_path(possible_uri, cache=None, local_path=os.curdir, timeout=None):
+    """
+    Retrieve data from a URI (or a URL). Return the input if it
+    cannot be parsed as a URI.
+
+    If ``possible_uri`` is a MAST URI, the file will be retrieved via
+    astroquery's `~astroquery.mast.ObservationsClass.download_file`.
+    If ``possible_uri`` is a URL, it will be retrieved via astropy with
+    `~astropy.utils.data.download_file`.
+
+    Parameters
+    ----------
+    possible_uri : str or other
+        This input will be returned without changes if it is not a string,
+        or if it is a local file path to an existing file. Otherwise,
+        it will be parsed as a URI. Local file URIs beginning with ``file://``
+        are not supported by this method – nor are they necessary, since string
+        paths without the scheme work fine! Cloud FITS are not yet supported.
+    cache: None, bool, or ``"update"``, optional
+        Cache file after download. If ``possible_uri`` is a
+        URL, ``cache`` may be a boolean or ``"update"``, see documentation for
+        `~astropy.utils.data.download_file` for details. If cache is None,
+        the file is cached and a warning is raised suggesting to set ``cache``
+        explicitly in the future.
+    local_path : str, optional
+        Save the downloaded file to this path. Default is to
+        save the file with its remote filename in the current
+        working directory. This is only used if data is requested
+        from `astroquery.mast`.
+    timeout : float, optional
+        If downloading from a remote URL, set the timeout limit for
+        remote requests in seconds (passed to
+        `~astropy.utils.data.download_file`).
+
+    Returns
+    -------
+    possible_uri : str or other
+        If ``possible_uri`` cannot be retrieved as a URI, returns the input argument
+        unchanged. If ``possible_uri`` can be retrieved as a URI, returns the
+        local path to the downloaded file.
+    """
+
+    if local_path == os.curdir and os.environ.get("JDAVIZ_START_DIR", ""):
```

Comment:
I don't think it matters what the `local_path` is when in this mode? You would want to grab the JDAVIZ_START_DIR regardless.
```suggestion
    if os.environ.get("JDAVIZ_START_DIR", ""):
```

Reply:
Ok, got it.